src/static_canvas.class.ts

```diff
@@ -1609,16 +1619,49 @@ import { pick } from './util/misc/pick';
     },
 
     /**
-     * Clears a canvas element and dispose objects
-     * @return {fabric.Canvas} thisArg
-     * @chainable
+     * Waits until rendering has settled to destroy the canvas
+     * @returns {Promise<boolean>} a promise resolving to `true` once the canvas has been destroyed or to `false` if the canvas has was already destroyed
+     * @throws if aborted by a consequent call
      */
     dispose: function () {
-      // cancel eventually ongoing renders
-      if (this.isRendering) {
-        fabric.util.cancelAnimFrame(this.isRendering);
-        this.isRendering = 0;
-      }
+      this.disposed = true;
+      return new Promise<boolean>((resolve, reject) => {
+        const task = () => {
+          this.destroy();
+          resolve(true);
+        }
+        task.kill = reject;
+        if (this.__cleanupTask) {
+          this.__cleanupTask.kill('aborted');
+        }
+        
+        if (this.destroyed) {
+          resolve(false);
+        }
```

Comment:
What are we doing with those?
i see you want to resolve with true/false in case this is an effective dispose or not.
But why? what does it give us this extra information?

Reply:
Perhaps someone needs to wait until dispose is done to trigger some task that shouldn't  start if not disposed